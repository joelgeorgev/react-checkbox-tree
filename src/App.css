.content {
    margin: 2%;
}

.content ul{
    list-style: none;
}

.content ul input{
    cursor: pointer;
}