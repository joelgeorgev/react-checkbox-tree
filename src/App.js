import React from 'react';
import 'tachyons/css/tachyons.min.css';

import { Intro, CheckboxTree } from './components';
import github from './assets/github.svg';

export class App extends React.PureComponent {
  render() {
    return (
      <div className='flex flex-column w-80 mw8 vh-100 center pv4'>
        <div className='flex flex-auto flex-column'>
          <Intro />
          <CheckboxTree />
        </div>
        <div className='self-center'>
          <a href='https://github.com/joelgeorgev/react-checkbox-tree'>
            <img src={github} alt='GitHub' />
          </a>
        </div>
      </div>
    );
  }
}