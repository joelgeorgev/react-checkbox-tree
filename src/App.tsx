import styled from 'styled-components'

import { Intro, CheckboxTree } from './components'
import data from './data/data.json'
import github from './assets/github.svg'

const Main = styled.main`
  display: flex;
  flex-direction: column;
  width: 80%;
  max-width: 64rem;
  height: 100vh;
  margin: 0 auto;
`

const Section = styled.section`
  display: flex;
  flex-direction: column;
  flex: 1;
  overflow: auto;
`

const Footer = styled.footer`
  align-self: center;
  margin: 2rem;
`

export const App = () => (
  <Main>
    <header>
      <Intro />
    </header>
    <Section>
      <article>
        <CheckboxTree data={data} />
      </article>
    </Section>
    <Footer>
      <a href='https://github.com/joelgeorgev/react-checkbox-tree'>
        <img src={github} alt='Go to GitHub repository page' />
      </a>
    </Footer>
  </Main>
)
