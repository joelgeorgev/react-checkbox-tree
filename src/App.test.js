import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import { App } from './App'

it('smoke test', () => {
  const container = document.createElement('div')
  ReactDOM.render(<App />, container)
})

it('snapshot test', () => {
  const tree = renderer.create(<App />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('state change test', () => {
  const container = document.createElement('div')
  renderer.act(() => {
    ReactDOM.render(<App />, container)
  })
  const inputElement = container.querySelectorAll('input')[2]
  renderer.act(() => {
    inputElement.dispatchEvent(new MouseEvent('click', { bubbles: true }))
  })
  expect(inputElement.checked).toBe(true)
})
