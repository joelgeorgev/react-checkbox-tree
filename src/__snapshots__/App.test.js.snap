// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<main
  role="main"
>
  <section
    className="flex flex-column w-80 mw8 vh-100 center pv5"
  >
    <div
      className="flex flex-auto flex-column"
    >
      <div>
        <h3>
          React Checkbox Tree
        </h3>
        <p>
          A react app showcasing a simple checkbox tree component. This project was created using
           
          <a
            href="https://github.com/facebookincubator/create-react-app"
          >
            create-react-app
          </a>
          .
        </p>
      </div>
      <ul
        className="list"
      >
        <li>
          <ul
            className="list"
          >
            <li>
              <label>
                <input
                  checked={false}
                  className="mr2 pointer"
                  onChange={[Function]}
                  type="checkbox"
                />
                Name
              </label>
            </li>
            <li>
              <label>
                <input
                  checked={false}
                  className="mr2 pointer"
                  onChange={[Function]}
                  type="checkbox"
                />
                Age
              </label>
            </li>
            <li>
              <label>
                <input
                  checked={false}
                  className="mr2 pointer"
                  onChange={[Function]}
                  type="checkbox"
                />
                Contact
              </label>
            </li>
            <li>
              <ul
                className="list"
              >
                <li>
                  <label>
                    <input
                      checked={false}
                      className="mr2 pointer"
                      onChange={[Function]}
                      type="checkbox"
                    />
                    Telephone
                  </label>
                </li>
                <li>
                  <ul
                    className="list"
                  >
                    <li>
                      <label>
                        <input
                          checked={false}
                          className="mr2 pointer"
                          onChange={[Function]}
                          type="checkbox"
                        />
                        Home
                      </label>
                    </li>
                    <li>
                      <label>
                        <input
                          checked={false}
                          className="mr2 pointer"
                          onChange={[Function]}
                          type="checkbox"
                        />
                        Work
                      </label>
                    </li>
                  </ul>
                </li>
                <li>
                  <label>
                    <input
                      checked={false}
                      className="mr2 pointer"
                      onChange={[Function]}
                      type="checkbox"
                    />
                    Mobile
                  </label>
                </li>
              </ul>
            </li>
            <li>
              <label>
                <input
                  checked={false}
                  className="mr2 pointer"
                  onChange={[Function]}
                  type="checkbox"
                />
                Address
              </label>
            </li>
            <li>
              <ul
                className="list"
              >
                <li>
                  <label>
                    <input
                      checked={false}
                      className="mr2 pointer"
                      onChange={[Function]}
                      type="checkbox"
                    />
                    House
                  </label>
                </li>
                <li>
                  <label>
                    <input
                      checked={false}
                      className="mr2 pointer"
                      onChange={[Function]}
                      type="checkbox"
                    />
                    Road
                  </label>
                </li>
                <li>
                  <label>
                    <input
                      checked={false}
                      className="mr2 pointer"
                      onChange={[Function]}
                      type="checkbox"
                    />
                    Zip Code
                  </label>
                </li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </div>
    <div
      className="self-center"
    >
      <a
        href="https://github.com/joelgeorgev/react-checkbox-tree"
      >
        <img
          alt="GitHub"
          src="github.svg"
        />
      </a>
    </div>
  </section>
</main>
`;
